### DISCUSSION TITLE
2.2.3-SNAPSHOT - Addressing problems with passwords as Strings

### DISCUSSION THREAD

**Discussion by jorabin** [jorabin/KeePassJava2#60] [Announcements]
I've done some work addressing problems with getting and setting protected values as strings and storing them as strings in the database. This has been subject of various discussions over a long long time including #55, #53 and #28. 

The code is on `develop` and there is a lengthy discussion of [protected property protection](https://github.com/jorabin/KeePassJava2/blob/develop/PropertyValueProtection.md) and the [readme](https://github.com/jorabin/KeePassJava2/blob/develop/readme.md) has been updated too.

The long and the short is that the Jackson implementation (released as 2.2.2, now updated) contains the enhancements, the other three database implementations do not, if you think there is a reason to keep any of them or that it would be worth implementing this enhancement let me know.

I'd really appreciate any and all feedback here on this direction.

Thank you!
Jo

Revision 2.2.3-SNAPSHOT containing these updates is available at the Sonatype repo:

        <groupId>org.linguafranca.pwdb</groupId>
        <artifactId>KeePassJava2</artifactId>
        <version>2.2.3-SNAPSHOT</version>

repository coordinates per the main POM. i.e.

         <repositories>
           <repository>
             <id>oss.sonatype.org-snapshot</id>
             <url>https://oss.sonatype.org/content/repositories/snapshots</url>
             <releases>
               <enabled>false</enabled>
             </releases>
             <snapshots>
               <enabled>true</enabled>
             </snapshots>
           </repository>
         </repositories>

**Comment by giusvale-dev** [comment]
Wow @jorabin! Well done! It's great work

**Reply by jorabin** [reply]
Thanks! I am working on some improvements to `PropertyValue.java` and will appreciate your review.

**Reply by jorabin** [reply]
I've just pushed some rather major changes to [PropertyValue.java](https://github.com/jorabin/KeePassJava2/blob/a6e64f3302ab96d070af9fb31567d9db28600c06/database/src/main/java/org/linguafranca/pwdb/PropertyValue.java) having discovered a few bugs and deficiencies in the original (yes, so soon). 

I am very keen that this gets close scrutiny and look forward to hearing feedback in this discussion.

Thanks

**Comment by jorabin** [comment]
Noting that 2.2.3 has now been released.
